src/app/static/src/app/components/adr/SelectDatasetModal.vue

```diff
@@ -1,45 +1,49 @@
-                 style="margin-top: 15px"
-                 id="fetching-datasets">
-                <loading-spinner size="xs"></loading-spinner>
-                <span v-translate="'loadingDatasets'"></span>
+        <template v-if="loading">
+            <p v-translate="'importingFiles'"></p>
+            <div class="text-center" id="loading-dataset">
+                <loading-spinner size="sm"></loading-spinner>
             </div>
-            <div v-if="fetchingError" id="fetch-error">
-                <div v-if="fetchingError.detail">{{fetchingError.detail}}</div>
-                <button @click="fetchDatasets"
-                        class="btn btn-red float-right"
-                        v-translate="'tryAgain'">
-                </button>
+        </template>
+        <template v-else>
+            <h4 v-translate="'browseADR'"></h4>
+            <div>
+                <label class="font-weight-bold" v-translate="'datasets'"></label>
+                <hint-tree-select id="datasetSelector"
+                                  :multiple="false"
+                                  :searchable="true"
+                                  :options="datasetOptions"
+                                  :placeholder="select"
+                                  :disabled="fetchingDatasets"
+                                  v-model="datasetId">
+                    <template v-slot:option-label="{node}">
+                        <label v-html="node.raw.customLabel">
+                        </label>
+                    </template>
+                </hint-tree-select>
+                <select-release :dataset-id="datasetId"
+                                :open="open"
+                                :dataset-type="datasetType"
+                                @selected-dataset-release="(newReleaseId) => datasetReleaseId = newReleaseId"
+                                @valid="(newValue) => valid = newValue">
+                </select-release>
+                <div :class="fetchingDatasets ? 'visible' : 'invisible'"
+                     style="margin-top: 15px"
+                     id="fetching-datasets">
+                    <loading-spinner size="xs"></loading-spinner>
+                    <span v-translate="'loadingDatasets'"></span>
+                </div>
+                <div v-if="fetchingError" id="fetch-error">
+                    <div v-if="fetchingError.detail">{{ fetchingError.detail }}</div>
+                    <button @click="fetchDatasets"
+                            class="btn btn-red float-right"
+                            v-translate="'tryAgain'">
+                    </button>
+                </div>
             </div>
-        </div>
-        <div class="text-center" v-if="loading" id="loading-dataset">
-            <loading-spinner size="sm"></loading-spinner>
-        </div>
-        <template v-slot:footer v-if="!loading && datasetId">
+        </template>
+        <template v-slot:footer v-if="!loading">
```

Comment:
wait, couldnt this be up with the `v-else` because that above template is already not loading? also do we not need to check if `datasetId` exists anymore?

Reply:
So this did not work for me inside the v-else. Got an error "ESLint: 'v-slot' directive must be owned by a custom element, but 'template' is not.(vue/ valid-v-slot)". So I think we need this to be nested directly in the modal, not inside another template.

We don't and actually was getting test failures due to this. Actually a bit confused how this was passing before, I possibly some regression from other changes in this PR but not sure exactly what. With the `datasetId` I could only close the dialog by picking a dataset. Pick a dataset, datasetId gets a value and now the footer with the buttons shows up. Which seemed a bit weird to me. We should be able to close the modal without selecting a dataset.

Reply:
so template with v-slot attribute is one was to define slots the other way is using the `<slot>` tag, and this way it doesnt complain about it:
```ts
            <slot name="footer">
                <div style="float: right;">
                    <button id="importBtn"
                            type="button"
                            :disabled="disableImport"
                            class="btn btn-red"
                            v-translate="'import'"
                            @click="$emit('confirmImport', datasetId!!, datasetReleaseId)">
                    </button>
                    <button type="button"
                            class="btn btn-white"
                            v-translate="'cancel'"
                            @click="$emit('closeModal')">
                    </button>
                </div>
            </slot>
```
you can put that at the end of that `template v-else`

i see the button is disabled if `datasetId` is falsy so fair enough, i think thats a sensible decision

Reply:
also sorry can replace `float: right;` inline style with class `"float-right"`